openedx/features/genplus_features/genplus/models.py

```diff
@@ -33,3 +67,41 @@ class GenUser(models.Model):
     school = models.ForeignKey(School, on_delete=models.SET_NULL, null=True)
     year_of_entry = models.CharField(max_length=32, null=True, blank=True)
     registration_group = models.CharField(max_length=32, null=True, blank=True)
+
+    @property
+    def is_student(self):
+        return self.role == self.STUDENT
+
+    @property
+    def is_teacher(self):
+        return self.role == self.TEACHING_STAFF
+
+
+class Teacher(models.Model):
+    user_profile = models.OneToOneField(GenUser, on_delete=models.CASCADE)
+    profile_image = models.ImageField(upload_to='gen_plus_teachers', null=True)
+    classes = models.ManyToManyField(Class, related_name='classes')
+
+
+class Student(models.Model):
+    user_profile = models.OneToOneField(GenUser, on_delete=models.CASCADE)
+    character = models.ForeignKey(Character,on_delete=models.SET_NULL, null=True)
+    on_board = models.BooleanField(default=False)
```

Comment:
Can we move this to GenUser because teacher also needs to be onboarded. Also we need to keep in mind that onboarding includes tour of platform which student and teacher can take again anytime from the profile